File: JavaScript/uri-1039.js
// URI - 1039: https://www.urionlinejudge.com.br/judge/pt/problems/view/1039

const input = require('fs').readFileSync('/dev/stdin', 'utf-8');
const lines = input.split('\n');

const numberOfLines = lines.length;
let data, result;

for (let i = 0; i < numberOfLines; i++) {
  data = getHunterAndFlowerData(lines.shift().split(' '));
  result = catchFireFlower(data.hunter, data.flower);
  console.log(result);
}

function getHunterAndFlowerData(line) {
  const hunter = {
    radius: parseInt(line[0]),
    coordinateX: parseInt(line[1]),
    coordinateY: parseInt(line[2])
  };
  const flower = {
    radius: parseInt(line[3]),
    coordinateX: parseInt(line[4]),
    coordinateY: parseInt(line[5])
  };
  return { hunter, flower };
}

function catchFireFlower(hunter, flower) {
  if (hunterRadiusIsGreaterThanEuclideanDistancePlusFlowerRadius(hunter, flower))
    return 'RICO';
  else
    return 'MORTO';
}

function hunterRadiusIsGreaterThanEuclideanDistancePlusFlowerRadius(hunter, flower) {
  return hunter.radius >= 
    (calculateEuclideanDistance(hunter.coordinateX, hunter.coordinateY, flower.coordinateX, flower.coordinateY) + flower.radius);
}

function calculateEuclideanDistance(x1, y1, x2, y2) {
  return Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
}

File: Ruby/uri-1039.rb
# frozen_string_literal: true

# URI - 1039: https://www.urionlinejudge.com.br/judge/pt/problems/view/1039

def take_fire_flower(values)
  if does_the_hunter_catch_the_flower?(values)
    'RICO'
  else
    'MORTO'
  end
end

def does_the_hunter_catch_the_flower?(values)
  values[0] >= calculate_euclidean_distance(values) + values[3]
end

def calculate_euclidean_distance(values)
  Math.sqrt(((values[4] - values[1])**2) + ((values[5] - values[2])**2))
end

while values = gets&.split(' ')&.map(&:to_i)
  puts take_fire_flower(values)
end
